src/components/Project/DnDColumnList.tsx

```diff
@@ -0,0 +1,146 @@
+import { useState, createContext } from "react";
+import styled from "styled-components";
+import { DragDropContext, Draggable, DraggableLocation, DragStart, Droppable, DropResult } from "react-beautiful-dnd";
+
+import { moveItemToAnotherList, reorderSameListItems } from "../../utils/dragAndDrop";
+
+import ColumnDefinition from "../../interfaces/Column";
+import Column from "../Column";
+
+export const GRABBED_ITEM_CATEGORY = {
+  COLUMN: "COLUMN",
+  ROW_TICKET: "ROW_TICKET",
+} as const;
+
+type GrabbedItemCategory = keyof typeof GRABBED_ITEM_CATEGORY;
+
+export const GrabbedItemContext = createContext<GrabbedItemCategory | null>(null);
+
+interface DnDColumnListProps {
+  projectColumns: ColumnDefinition[];
+}
+
+/**
+ * Panel of drag-and-drop-able columns. It has logic to;
+ * - move dragged item with the same list
+ * - move dragged item to another list
+ * - disable different item's drop -> e.g. when ticket is dragged, it can't be dropped upon column's drop area
+ * @param projectColumns - current focused project's list of column
+ */
+export default function DnDColumnList({ projectColumns }: DnDColumnListProps): JSX.Element {
```

Comment:
This component is the "central" place of DnD context. Feel free to ping me if you have any questions/concerns about the implementation around here